packages/contractkit/src/utils/signing-utils.ts

```diff
@@ -171,7 +171,13 @@ export function recoverTransaction(rawTx: string): [Tx, string] {
     data: rawValues[8],
     chainId,
   }
-  const signature = Account.encodeSignature(rawValues.slice(9, 12))
+  let r = rawValues[10]
+  let s = rawValues[11]
+  // Account.recover cannot handle canonicalized signatures
+  // A canonicalized signature may have the first byte removed if its value is 0
+  r = ensureLeading0x(trimLeading0x(r).padStart(64, '0'))
```

Comment:
I think you could also use `r = Bytes.pad(32, r)`
this is the code:
```
const pad = (l,hex) =>
  hex.length === l*2+2 ? hex : pad(l,"0x"+"0"+hex.slice(2));
```
It's less optimal, but it's a library that we are already using (both work for me, haha, your choice)

Reply:
I'm trying to reduce the usage of the eth-lib wherever possible due to the message you pointed out:
> please, don't use this lib
https://github.com/MaiaVictor/eth-lib

Reply:
https://github.com/celo-org/celo-monorepo/issues/3410